src/dispatcher/src/main/java/org/acme/ControllerResource.java

```diff
@@ -0,0 +1,51 @@
+package org.acme;
+
+import io.quarkus.funqy.knative.events.CloudEvent;
+import io.cloudevents.core.builder.CloudEventBuilder;
+import io.quarkus.funqy.Funq;
+import io.vertx.core.json.JsonObject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.logging.Logger;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.net.URI;
+import java.time.LocalDateTime;
+import java.util.UUID;
+
+public class ControllerResource {
+
+    Logger logger = Logger.getLogger(ControllerResource.class);
+
+    @ConfigProperty(name = "app.service.name", defaultValue = "app-service")
+    String serviceName;
+    @ConfigProperty(name = "app.revision.name", defaultValue = "app-revision")
+    String revisionName;
+
+    @Inject
+    @RestClient
+    EventNotifier eventNotifier;
+
+    @Funq
+    public String dispatchEvent(CloudEvent<JsonObject> cloudEvent) {
```

Comment:
Any reason for not using the same pattern as the LoaderFunction and model a ProducedEvent class to define the expected event?
Also, the dispatched event can be modelled as another class DispatchedEvent instead of a JsonObject, to better clarify the scope of the API

Reply:
69ca4b83e95058780b2dfa3fab213104663c319d